File: 323-NumberOfConnectedComponentsInAnUndirectedGraph.py
from collections import defaultdict, deque
class Solution(object):
    def countComponents(self, n, edges):
        """
        :type n: int
        :type edges: List[List[int]]
        :rtype: int
        """
        graph = defaultdict(list)
        visited = set()
        for e1, e2 in edges:
            graph[e1].append(e2)
            graph[e2].append(e1)
        count = 0
        for i in range(n):
            if i not in visited:
                queue = deque([])
                queue.append(i)
                while queue:
                    curr = queue.popleft()
                    for child in graph[curr]:
                        if child not in visited:
                            visited.add(child)
                            queue.append(child)
                count+=1
        return count

File: JavaScript/323-Number of Connected Components in an Undirected Graph.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number}
 */
var countComponents = function(n, edges) {
    let graph = new Map();
    let visited = new Array(n).fill(false);
    
    for (let i=0; i<n; i++){
        graph.set(i, []);
    }
    
    for (let [u,v] of edges){
        graph.set(u, graph.get(u).concat([v]) );
        graph.set(v, graph.get(v).concat([u]) );
    }
    
    let dfs = (start) =>{
        for (let nxt of graph.get(start)){
            if (!visited[nxt]){
                visited[nxt] = true;
                dfs(nxt);
            }
        }
    }
    
    let count = 0;
    
    for (let i=0; i<n; i++){
        if(!visited[i]){
            count++;
            dfs(i);
        }
    }
    
    return count;
};